fw/t/unit/test_http_parser_common.h

```diff
@@ -572,32 +554,39 @@ do_split_and_parse(int type, int chunk_mode)
  */
 static
 int __attribute__((unused))
-validate_data_fully_parsed(int type)
+validate_data_fully_parsed(int type, size_t sz_diff)
 {
 	TfwHttpMsg *hm = (type == FUZZ_REQ || type == FUZZ_REQ_H2)
 			? (TfwHttpMsg *)req
 			: (TfwHttpMsg *)resp;
 
+	size_t hm_exp_len = GET_FRAMES_TOTAL_SZ() - sz_diff;
+
 	EXPECT_EQ(hm->msg.len, hm_exp_len);
 	return hm->msg.len == hm_exp_len;
 }
 
-#define TRY_PARSE_EXPECT_PASS(type, chunk_mode)				\
-({									\
+#define __TRY_PARSE_EXPECT_PASS(type, sz_diff, chunk_mode)		\
+chunk_size_index = 0;							\
+while (({								\
 	int _err = do_split_and_parse(type, chunk_mode);		\
 	if (_err == TFW_BLOCK || _err == TFW_POSTPONE			\
-	    || !validate_data_fully_parsed(type))   			\
+	    || !validate_data_fully_parsed(type, sz_diff))		\
 		TEST_FAIL("can't parse %s (code=%d)\n",			\
 			  (type == FUZZ_REQ	    			\
 			   || type == FUZZ_REQ_H2			\
 			   ? "request" : "response"),			\
 			  _err);					\
 	__fpu_schedule();						\
 	_err == TFW_PASS;						\
-})
+}))
```

Comment:
There is no usage of the macro, when the `while` loop has body, so please reorganize into a normal loop with a body, not just do everything in the condition

Reply:
I didn't quite understand the comment. We must use that ugly construction in condition because in some cases we add check-block after implicit EXPECT-macro. Like this:
```
 	FOR_REQ_H2(
	    HEADERS_FRAME_BEGIN();
		// :method = "GET"
		HEADER(INDEX(2));
		// :scheme = "https"
		HEADER(INDEX(7));
		// :path = "/"
		HEADER(INDEX(4));
	    HEADERS_FRAME_END();
	)
	{
		EXPECT_EQ(req->method, TFW_HTTP_METH_GET);
		EXPECT_TFWSTR_EQ(&req->h_tbl->tbl[TFW_HTTP_HDR_H2_SCHEME],
				 ":scheme" "https");
		EXPECT_TFWSTR_EQ(&req->uri_path, "/");
	}
```
I can't take and move all the operations from the condition block to the loop body. So what do I need to do?

Reply:
I just didn't find a place where the macro has a body. If there is a place, where the macro has a body, then my comment is wrong